operator/src/test/java/org/keycloak/operator/ClusteringE2EIT.java

```diff
@@ -0,0 +1,201 @@
+        });
+    }
+
+    // local debug commands:
+    //    export TOKEN=$(curl --data "grant_type=password&client_id=token-test-client&username=test&password=test" http://localhost:8080/realms/token-test/protocol/openid-connect/token | jq -r '.access_token')
+    //
+    //    curl http://localhost:8080/realms/token-test/protocol/openid-connect/userinfo -H "Authorization: bearer $TOKEN"
+    //
+    //    example good answer:
+    //    {"sub":"b660eec6-a93b-46fd-abb2-e9fbdff67a63","email_verified":false,"preferred_username":"test"}
+    //    example error answer:
+    //    {"error":"invalid_request","error_description":"Token not provided"}
+    @Test
+    public void testKeycloakCacheIsConnected() {
+        // given
+        Log.info("Setup");
+        var kc = K8sUtils.getDefaultKeycloakDeployment();
+        var crSelector = k8sclient
+                .resources(Keycloak.class)
+                .inNamespace(kc.getMetadata().getNamespace())
+                .withName(kc.getMetadata().getName());
+        var targetInstances = 3;
+        kc.getSpec().setInstances(targetInstances);
+        k8sclient.resources(Keycloak.class).inNamespace(namespace).createOrReplace(kc);
+        var realm = k8sclient.resources(KeycloakRealmImport.class).inNamespace(namespace).load(getClass().getResourceAsStream("/token-test-realm.yaml"));
+        var realmImportSelector = k8sclient.resources(KeycloakRealmImport.class).inNamespace(namespace).withName("example-token-test-kc");
+        realm.createOrReplace();
+
+        Log.info("Waiting for a stable Keycloak Cluster");
+        Awaitility.await()
+                .atMost(10, MINUTES)
+                .pollDelay(5, SECONDS)
+                .ignoreExceptions()
+                .untilAsserted(() -> {
+                    Log.info("Checking realm import has finished.");
+                    CRAssert.assertKeycloakRealmImportStatusCondition(realmImportSelector.get(), KeycloakRealmImportStatusCondition.DONE, true);
+                    Log.info("Checking Keycloak is stable.");
+                    CRAssert.assertKeycloakStatusCondition(crSelector.get(), KeycloakStatusCondition.READY, true);
+                });
+
+        Log.info("Testing the Keycloak Cluster");
+        Awaitility.await().atMost(5, MINUTES).ignoreExceptions().untilAsserted(() -> {
+            // Get the list of Keycloak pods
+            var pods = k8sclient
+                    .pods()
+                    .inNamespace(namespace)
+                    .withLabels(Constants.DEFAULT_LABELS)
+                    .list()
+                    .getItems();
+
+            String token = null;
+            // Obtaining the token from the first pod
+            // Connecting using port-forward and a fixed port to respect the instance issuer used hostname
+            for (var pod: pods) {
+                Log.info("Testing Pod: " + pod.getMetadata().getName());
+                try (var portForward = k8sclient
```

Comment:
Please add some comment here for future generations with reasoning why we took this approach, why can't we use in-cluster curl.

Reply:
The comment at line 133 was intended for this purpose, do you mind to suggest a different wording in case?

Reply:
Nah, ok. Looking at it again and it seems sufficient. 👍